__tests__/endpoints.test.js

```diff
@@ -0,0 +1,395 @@
+            
+            for (let endpoint in endpoints) {
+             const endpointObject = endpoints[endpoint]
+             const length = Object.keys(endpointObject).length
+
+             switch(length) {
+                case 1:
+                  // code block
+                  expect(endpointObject).toMatchObject({
+                    description : expect.any(String),
+                })
+                  break;
+                case 2:
+                  // code block
+                  expect(endpointObject).toMatchObject({
+                    description : expect.any(String),
+                    queries : expect.any(Array),
+                })
+                  break;
+                case 3:
+                    expect(endpointObject).toMatchObject({
+                        description : expect.any(String),
+                        queries : expect.any(Array),
+                        exampleResponse : expect.any(Object),
+                    })
+                  break;
+                default:
+              }
+
+
+            }
+        })
+    })
+})
+
+    describe('Article Tests', () => {
+    describe('For retrieving an individual article', () => {
+    test('Status 200: GETs an Article when given an article ID which we have',()=>{
+        return request(app)
+        .get('/api/articles/3')
+        .expect(200)
+        .then((data)=>{    
+            const article = data.body
+            expect(article).toMatchObject({
+                article_id: expect.any(Number),
+                title : expect.any(String),
+                topic : expect.any(String),
+                author : expect.any(String),
+                body : expect.any(String),
+                created_at : expect.any(String),
+                votes : expect.any(Number),
+                article_img_url: expect.any(String)
+            })
+        })
+    })
+
+    test('Status 404: Gives an error message and a status 404 for non existant article',()=>{
+        return request(app)
+        .get('/api/articles/999')
+        .expect(404)
+        .then((data) => {
+            const err = data.body
+            expect(err).toMatchObject({ status: 404, msg: 'Not found' })
+        })
+    })
+
+    test('Status 400: Gives an error message for entering a parametric endpoint of wrong data type when trying to retrieve a specific article ',()=>{
+        return request(app)
+        .get('/api/articles/hello')
+        .expect(400)
+        .then((data) => {
+            const err = data.body
+            expect(err).toMatchObject({ status: 400, msg: 'Bad Request' })
+            
+        }) 
+    })
+    })
+
+    describe('gets the article along with a comment_count display number of comments article had', () => {
+        test('Status 200: sucesfully returns specified article along with a comment_count displaying the number of comments the article has',()=>{
+            return request(app)
+            .get('/api/articles/3/comment_count')
```

Comment:
This endpoint is incorrect. It should be `/api/articles/3` No need for the comment count in the url